test/compile-test.cc

```diff
@@ -192,6 +192,30 @@ TEST(compile_test, format_to_n) {
   EXPECT_STREQ("2a", buffer);
 }
 
+TEST(compile_test, format_std_error_code) {
+  EXPECT_EQ("generic:42",
+            fmt::format(FMT_COMPILE("{0}"),
+                        std::error_code(42, std::generic_category())));
+  EXPECT_EQ("system:42",
+            fmt::format(FMT_COMPILE("{0}"),
+                        std::error_code(42, std::system_category())));
+  EXPECT_EQ("system:-42",
+            fmt::format(FMT_COMPILE("{0}"),
+                        std::error_code(-42, std::system_category())));
+}
+
+TEST(compile_test, format_std_error_code_wide) {
+  EXPECT_EQ(L"generic:42",
+            fmt::format(FMT_COMPILE(L"{0}"),
+                        std::error_code(42, std::generic_category())));
+  EXPECT_EQ(L"system:42",
+            fmt::format(FMT_COMPILE(L"{0}"),
+                        std::error_code(42, std::system_category())));
+  EXPECT_EQ(L"system:-42",
+            fmt::format(FMT_COMPILE(L"{0}"),
+                        std::error_code(-42, std::system_category())));
+}
+
```

Comment:
No need to replicate the other formatting tests. Please remove these and add one check that parse is constexpr to the other test.